cmd/config/config.go

```diff
@@ -0,0 +1,97 @@
+package config
+
+import (
+  "os"
+  "strings"
+
+  "github.com/Originate/gt/cmd/util"
+)
+
+
+func GetMainBranch() string {
+  return getConfigurationValue("main-branch-name")
+}
+
+
+func GetParentBranch(branchName string) string {
+  return getConfigurationValue(branchName + ".parent")
+}
+
+
+func GetPerennialBranches() []string {
+  return strings.Split(getConfigurationValue("perennial-branch-names"), " ")
+}
+
+
+func GetPullBranchStrategy() string {
+  pullBranchStrategy := getConfigurationValueWithDefault("pull-branch-strategy", "rebase")
+  return pullBranchStrategy
+}
+
+
+func GetRemoteOriginUrl() string {
+  if os.Getenv("GIT_TOWN_ENV") == "test" {
+    mockRemoteUrl := getConfigurationValue("testing.remote-url")
+    if mockRemoteUrl != "" {
+      return mockRemoteUrl
+    }
+  }
+  return util.GetCommandOutput([]string{"git", "remote", "get-url", "origin"})
+}
+
+
+func GetRemoteUpstreamUrl() string {
+  return util.GetCommandOutput([]string{"git", "remote", "get-url", "upstream"})
+}
+
+
+func IsFeatureBranch(branchName string) bool {
+  return branchName != GetMainBranch() && !IsPerennialBranch(branchName)
+}
+
+
+func IsPerennialBranch(branchName string) bool {
+  perennialBranches := GetPerennialBranches()
+  return util.ContainsString(perennialBranches, branchName)
+}
+
+
+func HasRemoteOrigin() bool {
+  return GetRemoteOriginUrl() != ""
+}
+
+
+func HasRemoteUpstream() bool {
+  return GetRemoteUpstreamUrl() != ""
+}
+
+
+func ShouldHackPush() bool {
+  hackPushFlag := getConfigurationValueWithDefault("hack-push-flag", "true")
+  return hackPushFlag == "true"
+}
+
+
+func StoreParentBranch(branchName, parentBranchName string) {
+  storeConfigurationValue(branchName + ".parent", parentBranchName)
+}
+
+
+// Helpers
+
+
+func getConfigurationValue(key string) string {
+  return util.GetCommandOutput([]string{"git", "config", "git-town." + key})
+}
+
+func getConfigurationValueWithDefault(key, defaultValue string) string {
+  value := util.GetCommandOutput([]string{"git", "config", "git-town." + key})
+  if value == "" {
```

Comment:
I like to use guard clauses only for exceptional cases like error handling, and then return "legit" function results explicitly at the end, after the business logic of the function has run. I agree it's kind of debatable if the default value is an edge case or part of the happy path here, so either way is fine with me here.